lib/chore/strategies/worker/single_worker_strategy.rb

```diff
@@ -25,16 +33,14 @@ def stop!
       # single worker strategy, this should never be called if the worker is in
       # progress.
       def assign(work)
-        if workers_available?
-          begin
-            @worker = worker_klass.new(work, @options)
-            @worker.start
-            true
-          ensure
-            @worker = nil
-          end
-        else
-          Chore.logger.error { "#{self.class}#assign: single worker is unavailable, but assign has been re-entered: #{caller * "\n"}" }
+        return unless acquire_worker
```

Comment:
Should this be more like the forked code, where we use an if block?

I get where acquire_worker will block, and thus this shouldn't ever actually "return" here, but it seems to _imply_ that you could return early if the worker is not available.

Just a thought